runtime/src/server/middleware/get_page_handler.ts

```diff
@@ -225,7 +225,7 @@ export function get_page_handler(
 
 					props[`level${l++}`] = {
 						component: part.component,
-						props: preloaded[i + 1],
+						props: preloaded[i + 1] || {},
```

Comment:
I'm not sure but it seems like `null`, `0` and `""` are normal values for return from `preload` which are cut-off by this line

Reply:
May be comparison with `undefined` may be more suitable

Reply:
`preload` must always return an Object.

@cudr This might also error:
https://github.com/sveltejs/sapper/blob/6bf3dd04dd011db328e746bd3b7a3793f149ff20/runtime/src/server/middleware/get_page_handler.ts#L212

Reply:
Still can't understand why preload **must** return an object. From what I see the props are [just spread to components](https://github.com/sveltejs/sapper/blob/6bf3dd04dd011db328e746bd3b7a3793f149ff20/src/core/create_app.ts#L278). It shouldn't be error-prone 🤔

From where is [the error](https://github.com/sveltejs/sapper/issues/587) originated from?